Mithril update 2024-09-11 (#461)
IntersectMBO/cardano-updates: blog/2024-09-11-mithril.md

---
title: Mithril Team Update
slug: 2024-09-11-mithril
authors: jpraynaud
tags: [mithril]
hide_table_of_contents: false
---

## High level overview

> The Mithril team kept working on decentralizing the signature orchestration of the Mithril network. In this preliminary phase, they kept implementing a buffer store for individual signatures that may arrive before being processed by an aggregator. They also worked on a refactoring of the state machine of the signer. They worked on fixing panics occuring on the signer and aggregator when they operate roll-backs for Cardano transactions. Additionally, they made the pre-loading mechanism for the import of the Cardano transactions repeat forever in the signer.

Reviewer: ```suggestion
The Mithril team continued working on decentralizing the signature orchestration of the Mithril network. In this preliminary phase, they kept implementing a buffer store for individual signatures that may arrive before being processed by an aggregator. They also worked on refactoring the state machine of the signer and addressed panics occurring in both the signer and aggregator during rollbacks of Cardano transactions. Additionally, they modified the pre-loading mechanism for importing Cardano transactions, ensuring it repeats indefinitely in the signer.
```

> Finally, the team kept preparing the next distribution and they investigated some breaking change introduced in a dependency of the Hydra CI.

Reviewer: ```suggestion
Finally, the team continued preparing the next distribution and investigated a breaking change introduced in a Hydra CI dependency.
```
